enterprise/backend/src/metabase_enterprise/serialization/dump.clj

```diff
@@ -61,9 +61,16 @@
 
   Writes (even nested) yaml keys in a deterministic fashion."
   [filename obj]
-  (io/make-parents filename)
-  (spit filename (yaml/generate-string (serialization-deep-sort obj)
-                                       {:dumper-options {:flow-style :block :split-lines false}})))
+  (cond
+    (io/make-parents filename)
```

Comment:
Ca you explain why this is a test expression? I would expect `(not (.canWrite (io/file filename)))` to come first, and otherwise we run these two lines

Reply:
well `make-parents` promised to return `true` if it were able to create dirs and `false` otherwise. Turns out it returns `false` even if the directory exists already. 😁

Reply:
Where do you see that promise documented?

This is the docstring for `make-parents`:
```
  "Given the same arg(s) as for file, creates all parent directories of
   the file they represent."
```

I'd rather not rely on undocumented behaviour if we can avoid it

Reply:
```public boolean mkdirs()
Creates the directory named by this abstract pathname, including any necessary but nonexistent parent directories. Note that if this operation fails it may have succeeded in creating some of the necessary parent directories.
Returns:
true if and only if the directory was created, along with all necessary parent directories; false otherwise```